### PR TITLE
feat(server): [nan-906] run external client post connection scripts

### PR DESCRIPTION
**Pull request by khaliqgant**
## Describe your changes
Allow a customer to run a post installation script provided by them. Note that this task has a client deadline so has a bit more urgency. 

## Issue ticket number and link
NAN-906

## Checklist before requesting a review (skip if just adding/editing APIs & templates)
- [ ] I added tests, otherwise the reason is: 
- [ ] I added observability, otherwise the reason is:
- [ ] I added analytics, otherwise the reason is:

### CODE REVIEW

File: packages/shared/lib/clients/orchestrator.ts [hunk 1]
```diff
@@ -436,4 +437,193 @@ export class Orchestrator {
...
+                    .then(
+                        (res) => {
+                            if (res.isErr()) {
+                                logger.error(`Error: Execution '${executionId}' failed: ${stringifyError(res.error)}`);
+                            } else {
+                                logger.info(`Execution '${executionId}' executed successfully with result: ${res.value}`);
+                            }
+                        },
+                        (error) => {
+                            logger.error(`Error: Action '${executionId}' failed: ${stringifyError(error)}`);
```
Comment (khaliqgant) [inline packages/shared/lib/clients/orchestrator.ts]: Update from action

File: packages/shared/lib/clients/orchestrator.ts [hunk 2]
```diff
@@ -436,4 +437,193 @@ export class Orchestrator {
...
+                },
+                `actionName:${actionName}`
+            );
+            */
+
+            return Err(error);
+        } finally {
+            const endTime = Date.now();
+            const totalRunTime = (endTime - startTime) / 1000;
+            metrics.duration(metrics.Types.ACTION_TRACK_RUNTIME, totalRunTime);
```
Comment (khaliqgant) [inline packages/shared/lib/clients/orchestrator.ts]: Update from action

**Comment by TBonnin** [comment]
@khaliqgant sorry I haven't had time to review this PR yet. I'll do it first thing Friday morning

File: packages/shared/lib/db/migrations/20240529171723_add_post_connection_script.cjs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    return knex.schema.createTable(TABLE_NAME, function (table) {
+        table.increments('id').primary();
+        table.integer('config_id').unsigned().notNullable();
+        table.foreign('config_id').references('id').inTable('_nango_configs').onDelete('CASCADE');
+        table.text('name').notNullable();
+        table.text('file_location').notNullable();
+        table.string('version', 255).notNullable();
+        table.boolean('active').defaultTo(true);
+        table.timestamps(true, true);
+    });
```
Comment (bodinsamuel) [inline packages/shared/lib/db/migrations/20240529171723_add_post_connection_script.cjs]: needs an index config_id + name (where active = true)

**Review by bodinsamuel — APPROVED** [review]
Looks alright 👍🏻 
Some stuff are going to conflict with recent PRs

File: packages/orchestrator/lib/client.ts [hunk 4]
```diff
@@ -163,4 +176,16 @@ export class OrchestratorClient {
         };
         return this.execute(schedulingProps);
     }
+
+    public async executePostConnection(props: TExecutePostConnectionProps): Promise<TExecuteReturn> {
```
Comment (TBonnin) [inline packages/orchestrator/lib/client.ts]: you are gonna need to add a TaskPostConnection type and validation in orchestrator and add logic in jobs processor handler so the task can be processed
- https://github.com/NangoHQ/nango/blob/7774be4157ecd235fe737c6c662f368e309b121c/packages/orchestrator/lib/clients/types.ts#L35
- https://github.com/NangoHQ/nango/blob/7774be4157ecd235fe737c6c662f368e309b121c/packages/orchestrator/lib/clients/validate.ts#L24
- https://github.com/NangoHQ/nango/blob/7774be4157ecd235fe737c6c662f368e309b121c/packages/jobs/lib/processor/handler.ts#L6

You don't have to do it in this PR though.

**Review by TBonnin — APPROVED** [review]

File: packages/shared/lib/db/migrations/20240529171723_add_post_connection_script.cjs (hunk 3 above)
Comment (khaliqgant) [reply packages/shared/lib/db/migrations/20240529171723_add_post_connection_script.cjs]: https://github.com/NangoHQ/nango/pull/2225/files#diff-29f518b913a23b27a84df0fbacbd463e79873950b6b190add0575383f88d6d9d

File: packages/orchestrator/lib/client.ts (hunk 4 above)
Comment (khaliqgant) [reply packages/orchestrator/lib/client.ts]: Thanks, updated

**Pull request merged**